### ISSUE TITLE
system:masters tokens are not restricted by scopes

### ISSUE THREAD

**Issue by simo5** [openshift/origin#18922]
While trying to build an integration test for webhooks I came up to the fact that a user that uses a token with system:masters in the groups and scopes has scopes completely ignored and full access a system:masters.

This is in master

**Comment by simo5** [comment]
@deads2k @liggitt @enj comments ?

**Comment by liggitt** [comment]
Not surprising, but not really concerning. No OAuth-authenticated users can have that group (it's not a valid Group API object name)

**Comment by enj** [comment]
@deads2k made scopes authz come before system:masters, so it should deny.

**Comment by simo5** [comment]
@liggitt I was passing in that group from an external authorizer with no issue, so that statement is not fully true

**Comment by simo5** [comment]
@liggitt Note that passing system:masters as group is not a problem, but I remember discussing the authorization pipeline with @deads2k on a recent refactor and the scope authorizer was explicitly put before the system:masters check in order to allow using scopes in that case too

**Comment by deads2k** [comment]
> @liggitt Note that passing system:masters as group is not a problem, but I remember discussing the authorization pipeline with @deads2k on a recent refactor and the scope authorizer was explicitly put before the system:masters check in order to allow using scopes in that case too

Yeah, I'm surprised it didn't work.  It didn't work before, so we haven't regressed and can fix it later, but I did swizzle the order to try to deny early:

https://github.com/openshift/origin/blob/master/pkg/cmd/server/origin/authorizer.go#L37-L46

**Comment by simo5** [comment]
@deads2k that code is exactly what I was reminding, so we need to figure out at some point why it is not working.
I am not concerned this is a regression or anything or I would have made it a P0 and a Security issue :-)

**Comment by enj** [comment]
See https://github.com/openshift/origin/pull/18966#issuecomment-373743000 for an update on this.

**Comment by simo5** [comment]
This was by design and the "bug" was fixed in a different way

**Comment by enj** [comment]
The underlying issue of not being able to scope system:masters still remains right?  Or is there some changes in the upstream loopback token?

/remove-lifecycle stale
/lifecycle frozen

**Comment by simo5** [comment]
My understanding is that system:master is and will remain unscoped, but I'll defer to anyone that want sto do otherwise I guess.
